conda-store-server/conda_store_server/orm.py

```diff
@@ -225,15 +225,20 @@ def build_key(self):
         The build key should be a key that allows for the environment
         build to be easily identified and found in the database.
         """
-        datetime_format = "%Y%m%d-%H%M%S-%f"
-        return f"{self.specification.sha256}-{self.scheduled_on.strftime(datetime_format)}-{self.id}-{self.specification.name}"
+        hash = self.specification.sha256[:4]
+        timestamp = int(self.scheduled_on.timestamp())
+        id = self.id
+        name = self.specification.name[:16]
+        return f"{hash}-{timestamp}-{id}-{name}"
```

Comment:
This could be prefixed with `v2-` (or similar) to allow for multiple versions with 3 `-` delimiters right away. But I'm in favor of adding this in v3 or later if there are clashes. The formats are distinct now and we save 3 chars by not prefixing here, see `parse_build_key`.

Reply:
Nope. Cannot rely on dashes since name might contain them. Will use the hash size to differentiate between versions.